packages/ui/src/atom/circular-loader-atom/circular-loader-atom.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <field-atom
+    :class="themeCssClass"
+    :style="sizeStyle"
+  >
+    <svg
+      class="circular-loader-atom-back"
+      :style="sizeStyle"
+    >
+      <circle
+        :r="circleAttrs.r"
+        :cx="circleAttrs.cx"
+        :cy="circleAttrs.cy"
+        :stroke-width="lineThickness"
+      />
+    </svg>
+    <svg
+      class="circular-loader-atom-over"
+      :style="{ ...sizeStyle, strokeDasharray }"
+    >
+      <circle
+        :r="circleAttrs.r"
+        :cx="circleAttrs.cx"
+        :cy="circleAttrs.cy"
+        :stroke="circleAttrs.stroke"
+        :stroke-width="lineThickness"
+      />
+    </svg>
+    <slot />
+  </field-atom>
+</template>
+
+<script>
+import ThemeMixin from '@/mixins/ThemeMixin';
+import FieldAtom from '@/atom/field-atom/field-atom';
+
+// For initial state progress must show `action` like something is processing.
+// But if progress have `0` value, circle will be static and '0%' title.
+// Soo default behavior is always show animated circle without label
+// and if progress more than `1`, show progress title
+const MIN_PROGRESS_VALUE = 1;
+const DEFAULT_VIEW_VALUE = 2;
+const MAX_VALUE = 100;
+
+export default {
+  name: 'VProgressCircle',
+
+  props: {
+    progress: {
+      type: [Number, String],
+      default: 0,
+    },
+
+    animated: {
+      type: Boolean,
+      default: true,
+    },
+
+    size: {
+      type: [Number, String],
+      default: 82,
+    },
+
+    stroke: {
+      type: String,
+      default: '#6e32c9',
+    },
+
+    lineThickness: {
+      type: [Number, String],
+      default: 4,
+    },
+  },
+
+  computed: {
+    themeCssClass() {
+      return {
+        'circular-loader-atom': true,
+        animated: this.animated,
+      };
+    },
+
+    sizeStyle() {
+      return {
+        width: `${this.size}px`,
+        height: `${this.size}px`,
+      };
+    },
+
+    radius() {
+      return this.size / 2 - this.lineThickness / 2;
+    },
+
+    circleAttrs() {
+      const { size, radius, stroke } = this;
+
+      return {
+        r: radius,
+        cx: `${size / 2}px`,
+        cy: `${size / 2}px`,
+        stroke,
+      };
+    },
+
+    strokeDasharray() {
+      const { radius, progressValue } = this;
+      const circmference = radius * 2 * Math.PI;
+
+      return `${(circmference / 100) * progressValue} ${circmference}`;
+    },
+
+    progressText() {
+      return `${this.progress}%`;
+    },
+
+    progressValue() {
```

Comment:
progress value calculation should be in `VProgressCircle`, but not in current atom.
atom should only render circle and no need to know about any progress

Reply:
![image](https://user-images.githubusercontent.com/13414205/85124450-9e2bc900-b232-11ea-8f66-810f4860da4d.png)

We should use it for correct rendering

Reply:
in `strokeDasharray` you can use directly `this.progress`

this progress value should be only in `VProgressCircle`

or what is the reason for stay it in current atom?